File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:


        seen = {}
        for num in nums:
            seen[num] = False
        

        longest = 0

        for idx, num in enumerate(nums):

            if seen[num]:
                continue
            left, right = num - 1, num + 1
            while right in seen:
                seen[right] = True
                right += 1
            
            while left in seen:
                seen[left] = True
                left -= 1
            
            longest = max(longest, right - left - 1)
        
        return longest
        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int res = 1;
        if (nums.size() == 0) return 0;
        unordered_map<int, bool> seen;
        for (auto num: nums) seen[num] = false;

        for (auto num: nums) {
            if (seen[num]) continue;
            seen[num] = true;

            int left = num - 1;
            while (seen.find(left) != seen.end()) {
                seen[left] = true;
                left--;
            }
            int right = num + 1;
            while (seen.find(right) != seen.end()) {
                seen[right] = true;
                right++;
            }
            res = max(res, right - left - 1);
        }
        return res;
    }
};